### PR TITLE
Captain's Charter/Captain can name the station

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Allows the captain to name the station if he uses the charter within five minutes of round start.

It spawns in his bag.

> no kor put it in his office

If I put it in his office then I'm going to put a mindrole check to make sure it's the captain using it. This is meant to be a perk for playing one of the more stressful roles on station, not more loot for assistants when no captain signs up.

And if Chaplain swords taught me anything it's that players are suckers for silly customization perks that make the job feel special.

:cl: Kor
rscadd: The captain now spawns with the station charter, which allows him to name the station.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
Welcome to Reactive Teleportation Armor 13.

**Comment by KorPhaeron** [comment]
I'll job ban people who make bad names sorry Kevinz, immersion must be preserved.

**Comment by Iamgoofball** [comment]
The Wall #13

File: code/game/gamemodes/nuclear/nuclear_challenge.dm [hunk 1]
```diff
@@ -49,6 +49,5 @@
 
 
 #undef CHALLENGE_TELECRYSTALS
-#undef CHALLENGE_TIME_LIMIT
```
Comment (Shadowlight213) [inline code/game/gamemodes/nuclear/nuclear_challenge.dm]: Uhhhh. Intentional?

Comment (KorPhaeron) [inline code/game/gamemodes/nuclear/nuclear_challenge.dm]: Yes

Comment (Shadowlight213) [inline code/game/gamemodes/nuclear/nuclear_challenge.dm]: Yes as in intentional? How so?

**Comment by Jordie0608** [comment]
> I'll job ban people who make bad names sorry Kevinz, immersion must be preserved.

This will need to be clear to them before the inevitable. And we'll need an admin verb to change it back for when they do so anyway.

Also changelog.

File: code/game/gamemodes/nuclear/nuclear_challenge.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/game/gamemodes/nuclear/nuclear_challenge.dm]: I reuse the define for the charter

Comment (Shadowlight213) [inline code/game/gamemodes/nuclear/nuclear_challenge.dm]: Ah. I see

**Comment by PKPenguin321** [comment]
One Day While Andy Was 13

**Comment by TechnoAlchemisto** [comment]
I'll miss the random generation

File: code/game/objects/items/charter.dm [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+/obj/item/station_charter/attack_self(mob/living/user)
+	if(used)
+		user << "The station has already been named."
+		return
+	used = TRUE
+	if(world.time > CHALLENGE_TIME_LIMIT) //5 minutes
+		user << "The crew has already settled into the shift. It probably wouldn't be good to rename the station right now."
+		return
+
+	var/new_name = reject_bad_name( input(user, "What do you want to name [station_name()]? Keep in mind particularly terrible names may attract the attention of your employers.")  as text|null )
```
Comment (AnturK) [inline code/game/objects/items/charter.dm]: You might want to allow numbers and special characters here.

Comment (KorPhaeron) [inline code/game/objects/items/charter.dm]: oops

**Comment by coiax** [comment]
Why not just put a check in the New() that if it's five minutes since the station started, it just deletes itself. Then the captain isn't left with a random wizard scroll in their backpack that doesn't do anything.

**Comment by Bawhoppen** [comment]
@Jordie0608 there is a rule to use common sense is there not?

**Comment by Jordie0608** [comment]
There is. And yesterday I saw a captain named "Noadmin Stimeto Griff". Draw your own conclusion.

File: code/modules/admin/secrets.dm [hunk 3]
```diff
@@ -145,6 +146,13 @@
 				for(var/datum/disease/D in SSdisease.processing)
 					D.cure(D)
 
+		if("reset_name")
+			if(!check_rights(R_ADMIN))
+				return
+			world.name = "Space Station 13"
```
Comment (Jordie0608) [inline code/modules/admin/secrets.dm]: This should either use the initial name of the station or be a "Change Station Name" verb input() which defaults to it.

File: code/modules/admin/secrets.dm [hunk 4]
```diff
@@ -145,6 +146,13 @@
...
+			log_admin("[key_name(usr)] reset the station name.")
+			message_admins("<span class='adminnotice'>[key_name_admin(usr)] reset the station name.</span>")
```
Comment (Jordie0608) [inline code/modules/admin/secrets.dm]: Don't need span class in message_admins()

File: code/modules/admin/secrets.dm (hunk 3 above)
Comment (KorPhaeron) [inline code/modules/admin/secrets.dm]: If you captaim station names will be bad jusy wait until you see admins doing it

**Comment by Durkel** [comment]
I look forward to serving on NIGGER NIGGER NIGGER and MEMES station.

File: code/game/objects/items/charter.dm [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+/obj/item/station_charter/attack_self(mob/living/user)
+	if(used)
+		user << "The station has already been named."
+		return
+	used = TRUE
+	if(world.time > CHALLENGE_TIME_LIMIT) //5 minutes
+		user << "The crew has already settled into the shift. It probably wouldn't be good to rename the station right now."
+		return
+
+	var/new_name = input(user, "What do you want to name [station_name()]? Keep in mind particularly terrible names may attract the attention of your employers.")  as text|null
```
Comment (FlavoredCactus) [inline code/game/objects/items/charter.dm]: Sorry, I'm late on this. input() is unsafe, use stripped_input()

Comment (Iamgoofball) [inline code/game/objects/items/charter.dm]: dude im embedding a pornhub link as a station nname now dont take this away from me

**Pull request merged**